packages/sdk/src/CLRunner.ts

```diff
@@ -1693,6 +1716,29 @@ export class CLRunner {
         }
     }
 
+    private simulateEffectsOfMockResultOnMemory(entities: CLM.EntityBase[], callbackResult: CLM.CallbackResult, memoryManager: ClientMemoryManager) {
+        Object.entries(callbackResult.entityValues)
+            .forEach(([entityKey, entityValue]) => {
```

Comment:
Could you some more detailed commenting here.  Not clear to my why key is either the entityName or or and entityId (specifically the negative entityId).   With no context hard to tell why there are multiple keys of different types.

Reply:
> With no context hard to tell why there are multiple keys of different types.

That's why I added the comment saying why it checks both types.  Added more detail explaining what those types are though.

Reply:
Oh, I see, now. The comment appeared on the line above and didn't show the `negativeId` part. I was confused about that since you can't set negative entities in mock results. Negative entities is available for labels to indicate the removal of something from from positive entities values list.  Those negative entities shouldn't have any values. Makes think the memory manager should protect against this somehow, but that's unrelated.  I'm guessing I was typing "entity" and missed the 'e' and typed 'n' then autocompleted for negativeId. Updated to be correct `entityId`